contrib/epee/include/misc_language.h

```diff
@@ -128,6 +128,59 @@ namespace misc_utils
 
   }
 
+  template<typename T>
+  T factorial(T n, T s = 0)
+  {
+    static_assert(std::is_integral<T>::value, "Integral type required.");
+
+    if (s > n || n < 0 || s < 0)
+      return 0;
+
+    // n! / s!
+    // s as in 'stop'
+    T r{1};
+    while (n > s)
+    { 
+      if ((std::numeric_limits<T>::max() / r) < n)
+        return 0;
+
+      r *= n;
+      n--;
+    }
+
+    return r;
+  }
+
+  template<typename T>
+  T n_choose_k(T n, T k)
+  {
+    static_assert(std::is_integral<T>::value, "Integral type required.");
+
+    // n choose k = n! / (k! * (n - k)!)
+    if (n < 0 || k < 0)
+      return 0;
+    else if (n < k)
+      return 1;
+    else if (k >= n - k)
+    {
+      // (n! / k!) / (n - k)!
+      T numerator{factorial<T>(n - k)};
+      if (numerator == 0)
+        return 0;
+      else
+        return factorial<T>(n, k) / numerator;
```

Comment:
So if this overflow its `return 0;` ... hmm. I suppose that makes sense as an error case value, a comment br both functions may help with that (check for `0` as its the error case).